crates/typst-library/src/layout/place.rs

```diff
@@ -95,7 +95,10 @@ impl Layout for PlaceElem {
         let float = self.float(styles);
         let alignment = self.alignment(styles);
 
-        if float && alignment.map_or(false, |align| align.y() == Some(VAlign::Horizon)) {
+        if float
+            && alignment
+                .map_or(false, |align| matches!(align.y(), None | Some(VAlign::Horizon)))
```

Comment:
maybe we should invert the condition here to include the things bail! wants. basically
`matches!(alignment, Smart::Auto | Smart::Custom(Some(VAlign::Top | VAlign::Bottom)))`. Would make things clearer I think.